pkg/controller/atlasproject/cloud_provider_access.go

```diff
@@ -33,7 +33,16 @@ func ensureProviderAccessStatus(ctx context.Context, customContext *workflow.Con
 func syncProviderAccessStatus(ctx context.Context, customContext *workflow.Context, specs []v1.CloudProviderAccessRole, statuses []status.CloudProviderAccessRole, groupID string) (workflow.Result, status.ConditionType) {
 	client := customContext.Client
 	logger := customContext.Log
-	specToStatusMap := checkStatuses(specs, statuses)
+	specToStatusMap, haveDuplicate, cantMatch := checkStatuses(specs, statuses)
+	if haveDuplicate {
+		return workflow.Terminate(workflow.ProjectCloudAccessRolesIsNotReadyInAtlas, "some roles contains same ARN value"), status.CloudProviderAccessReadyType
+	}
+	if cantMatch {
+		return workflow.Terminate(workflow.ProjectCloudAccessRolesIsNotReadyInAtlas, "More than one new role "+
+			"with ARN may correspond to an existing empty role. Keep only one new role. If one of the new roles has been "+
```

Comment:
@Sugar-pack consider to change the message: "More than one new role with ARN may correspond to an existing empty role. Keep only one new role containing data from the status field and delete all other roles. You can add them again after authorization is complete."

Reply:
Done